src/modules/Verifications/containers/ReusableList/useReusableList.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  deleteVerificationAdaptor,
+  getReusableVerificationsAdaptor,
+  ReusableVerification,
+  ReusableVerificationsRes,
+  SingleUseVerificationsRes,
+  Verification,
+  VerificationsRes,
+} from 'src/core/adaptors';
+
+export const useReusableList = (
+  list: ReusableVerification[],
+  setList: (val: ReusableVerification[]) => void,
+  totalItems: number,
+  setOpenModal: (val: { name: 'delete' | 'copy' | 'history'; open: boolean }) => void,
+  selectedId?: string,
+) => {
+  const { t: translate } = useTranslation();
+
+  const [page, setPage] = useState(1);
+  const PER_PAGE = 10;
+  const [total, setTotal] = useState(Math.ceil(totalItems / PER_PAGE));
+
+  const fetchMore = async () => {
+    const res = await getReusableVerificationsAdaptor(page, PER_PAGE);
+    if (!res.data) return;
+    setList(res.data.items);
+    setTotal(Math.ceil(res.data.totalCount / PER_PAGE));
+  };
+
+  useEffect(() => {
+    fetchMore();
+  }, [page]);
+
+  const handleDelete = async () => {
+    if (!selectedId) return;
+    const res = await deleteVerificationAdaptor(selectedId);
+    if (res.error) return;
+    const newPage = list.length === 1 && page > 1 ? page - 1 : page;
+    const verificationRes = await getReusableVerificationsAdaptor(newPage, PER_PAGE);
+    if (verificationRes.error) return;
```

Comment:
why didn't you use your `fetchMore()` function here?
these two lines + line-45 has same functionality like `fetchMore()` function